File: compatibility-tests/basic-java8/gradle-java/src/test/java/ScenarioTest.java
import org.tabletest.junit.Scenario;
import org.tabletest.junit.TableTest;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class ScenarioTest {
    @TableTest({
        "Scenario | a | b | sum?",
        "Adding   | 1 | 2 | 3",
        "Zero sum | 0 | 0 | 0"
    })
    void scenario_parameter(@Scenario String scenario, int a, int b, int sum) {
        assertEquals(sum, a + b);
    }
}

File: compatibility-tests/basic/gradle-kotlin/src/test/kotlin/ScenarioTest.kt
import org.tabletest.junit.Scenario
import org.tabletest.junit.TableTest
import kotlin.test.assertEquals

class ScenarioTest {
    @TableTest("""
        Scenario | a | b | sum?
        Adding   | 1 | 2 | 3
        Zero sum | 0 | 0 | 0
        """)
    fun scenario_parameter(@Scenario scenario: String, a: Int, b: Int, sum: Int) {
        assertEquals(sum, a + b)
    }
}
